core/src/main/scala/org/http4s/blaze/pipeline/stages/TimeoutStageBase.scala

```diff
@@ -52,6 +61,11 @@ abstract class TimeoutStageBase[T](timeout: Duration, exec: TickWheelExecutor) e
   final protected def cancelTimeout(): Unit = setAndCancel(null)
 
   final protected def startTimeout(): Unit = resetTimeout()
-
-
 }
+
+private object TimeoutStageBase {
+  // Represents the situation where the pipeline has been closed.
+  val closedTag = new Cancellable {
```

Comment:
Is this intentionally not using `Cancellable.noopCancel`?  Is it the reference equality check in `setAndCancel`?  That should work, but, ick.  How about a case object that's Cancellable?